src/private_cuda.h

```diff
@@ -128,4 +140,14 @@ struct _gpukernel {
 #endif
 };
 
+struct _gpucomm {
+  cuda_context* ctx;  // Start after the context
+// #ifdef WITH_CUDA_NCCL
+  ncclComm_t c;
+// #endif  // WITH_CUDA_NCCL
```

Comment:
Allocating static array inside of a `gpucontext` and calling always `ncclGetErrorString(...)`?  Do we want to avoid the macros inside `cuda_context` for sure?